### ISSUE TITLE
bug: Executing script "mobile:swipe" results in a stale element error

### ISSUE THREAD

**Issue by grant-mccarriagher** [appium/appium#19739]
### Do I have the most recent component updates?

- [X] I use the most recent available driver/plugin and server versions

### Is the component officially supported by the Appium team?

- [X] I have verified the component repository is present under the Appium organization in GitHub

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

We recently updated Appium and xcuitest-driver to the latest versions, and noticed some "mobile:swipe" scripts have started to fail. Here's an example command:

```
[HTTP] --> POST /session/5d46e3fe-2f38-4fc0-bc7e-6ff8fb24f402/execute/sync
[HTTP] {"script":"mobile:swipe","args":[{"direction":"up","velocity":100}]}
```

Please see "Further Information" at the bottom of this post for the full snippet of log.

### Expected Behavior

Appium is able to execute the scroll without a stale element error.

### Minimal Reproducible Example

This seems to happen on any swipe script we do in our app. I haven't tested any other apps. If this isn't reproducible, please let me know.

### Environment

- Operating system: macOS
- Appium server version (output of `appium --version`): 2.4.1
- Appium driver(s) and their version(s): xcuitest 5.15.0
- Appium plugin(s) and their version(s):
- Node.js version (output of `node --version`): 16.20
- `npm` version (output of `npm --version`): 8.19.4
- Last component(s) version which did _not_ exhibit the problem: xcuitest 4.35 was the previous version we were on
- Platform and version under test: iOS 17.0
- Real device or emulator/simulator: Real

### Further Information

```
[HTTP] --> POST /session/5d46e3fe-2f38-4fc0-bc7e-6ff8fb24f402/execute/sync
[HTTP] {"script":"mobile:swipe","args":[{"direction":"up","velocity":100}]}
[XCUITestDriver@d04f (5d46e3fe)] Calling AppiumDriver.execute() with args: ["mobile:swipe",[{"direction":"up","velocity":100}],"5d46e3fe-2f38-4fc0-bc7e-6ff8fb24f402"]
[XCUITestDriver@d04f (5d46e3fe)] Executing command 'execute'
[XCUITestDriver@d04f (5d46e3fe)] Waiting up to 0 ms for condition
[XCUITestDriver@d04f (5d46e3fe)] Matched '/element' to command name 'findElement'
[XCUITestDriver@d04f (5d46e3fe)] Proxying [POST /element] to [POST http://127.0.0.1:8100/session/A4E57B8F-D660-4194-A96C-64D9B0516885/element] with body: {"using":"class name","value":"XCUIElementTypeApplication"}
[Xcode]     t =    90.37s Get all elements bound by accessibility element for: Descendants matching type Application
[Xcode]
[Xcode]     t =    90.37s     Requesting snapshot of accessibility hierarchy for app with pid 523
[Xcode]
[Xcode]     t =    90.40s     Find: Descendants matching type Application
[Xcode]
[Xcode]     t =    90.41s Requesting snapshot of accessibility hierarchy for app with pid 523
[Xcode]
[XCUITestDriver@d04f (5d46e3fe)] Got response with status 200: {"value":{"ELEMENT":"01000000-0000-0000-0B02-000000000000","element-6066-11e4-a52e-4f735466cecf":"01000000-0000-0000-0B02-000000000000"},"sessionId":"A4E57B8F-D660-4194-A96C-64D9B0516885"}
[XCUITestDriver@d04f (5d46e3fe)] Proxying to WDA with an unknown route: POST /wda/element/01000000-0000-0000-0B02-000000000000/swipe
[XCUITestDriver@d04f (5d46e3fe)] Proxying [POST /wda/element/01000000-0000-0000-0B02-000000000000/swipe] to [POST http://127.0.0.1:8100/session/A4E57B8F-D660-4194-A96C-64D9B0516885/wda/element/01000000-0000-0000-0B02-000000000000/swipe] with body: {"direction":"up","velocity":100}
[XCUITestDriver@d04f (5d46e3fe)] Got response with status 404: {"value":{"error":"stale element reference","message":"The element identified by \"01000000-0000-0000-0B02-000000000000\" is either not present or it has expired from the internal cache. Try to find it again","traceback":"(\n\t0   CoreFoundation                      0x00000001b0caa5ec CEB74A0C-5EA4-3F5A-9DE8-73F0DAD2F4CB + 972268\n\t1   libobjc.A.dylib                     0x00000001a8fdfc00 objc_exception_throw + 60\n\t2   WebDriverAgentLib                   0x00000001008cdc00 -[FBElementCache elementForUUID:resolveForAdditionalAttributes:andMaxDepth:] + 488\n\t3   WebDriverAgentLib                   0x00000001008cd9ec -[FBElementCache elementForUUID:] + 76\n\t4   WebDriverAgentLib                   0x00000001008d695c +[FBElementCommands handleSwipe:] + 176\n\t5   WebDriverAgentLib                   0x0000000100866108 -[FBRoute_TargetAction mountRequest:intoResponse:] + 168\n\t6   WebDriverAgentLib                   0x0000000100850064 __37-[FBWebServer registerRouteHandlers:]_block_invoke + 404\n\t7   WebD...
[W3C] Matched W3C error code 'stale element reference' to StaleElementReferenceError
[XCUITestDriver@d04f (5d46e3fe)] Encountered internal error running command: StaleElementReferenceError: The element identified by "01000000-0000-0000-0B02-000000000000" is either not present or it has expired from the internal cache. Try to find it again
[XCUITestDriver@d04f (5d46e3fe)]     at errorFromW3CJsonCode (/Users/grantmccarriagher/.appium/node_modules/appium-xcuitest-driver/node_modules/@appium/base-driver/lib/protocol/errors.js:1085:25)
[XCUITestDriver@d04f (5d46e3fe)]     at ProxyRequestError.getActualError (/Users/grantmccarriagher/.appium/node_modules/appium-xcuitest-driver/node_modules/@appium/base-driver/lib/protocol/errors.js:954:14)
[XCUITestDriver@d04f (5d46e3fe)]     at JWProxy.command (/Users/grantmccarriagher/.appium/node_modules/appium-xcuitest-driver/node_modules/@appium/base-driver/lib/jsonwp-proxy/proxy.js:353:19)
[XCUITestDriver@d04f (5d46e3fe)]     at runMicrotasks (<anonymous>)
[XCUITestDriver@d04f (5d46e3fe)]     at processTicksAndRejections (node:internal/process/task_queues:96:5)
[XCUITestDriver@d04f (5d46e3fe)]     at XCUITestDriver.proxyCommand (/Users/grantmccarriagher/.appium/node_modules/appium-xcuitest-driver/lib/commands/proxy-helper.js:109:35)
[XCUITestDriver@d04f (5d46e3fe)]     at XCUITestDriver.mobileSwipe (/Users/grantmccarriagher/.appium/node_modules/appium-xcuitest-driver/lib/commands/gesture.js:350:12)
[XCUITestDriver@d04f (5d46e3fe)]     at XCUITestDriver.executeMethod (/Users/grantmccarriagher/.nvm/versions/node/v16.20.0/lib/node_modules/appium/node_modules/@appium/base-driver/lib/basedriver/commands/execute.ts:36:12)
[XCUITestDriver@d04f (5d46e3fe)]     at XCUITestDriver.execute (/Users/grantmccarriagher/.appium/node_modules/appium-xcuitest-driver/lib/commands/execute.js:118:14)
[HTTP] <-- POST /session/5d46e3fe-2f38-4fc0-bc7e-6ff8fb24f402/execute/sync 404 107 ms - 1620
[HTTP]
```

**Comment by mykola-mokhnach** [comment]
Please upgrade xcuitest driver to v6 and try again. I have improved the way these gestures work there

**Comment by grant-mccarriagher** [comment]
On xcuitest 6.0.0, no longer seeing this error on scrolls. Thanks @mykola-mokhnach

**Comment by DipanGarg** [comment]
Hi @mykola-mokhnach could you please check, in case we introduced https://github.com/appium/appium/issues/19816 as part of the above changes. I did some back tracking and can confirm 19816 is present since we merged these changes(v6.0.0). It was working fine till xcuitestdriver@5.16.1 version.
